vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,674 @@
+   */
+  private void repaintArea() {
+    map.repaint(new Rectangle(
+            (int)(clickPoint.x - circleSize / 2.0 - 2 * STROKE * os_scale),
+            (int)(clickPoint.y - circleSize / 2.0 - 2 * STROKE * os_scale),
+            (int)(circleSize + 4 * STROKE * os_scale + 0.5),
+            (int)(circleSize + 4 * STROKE * os_scale + 0.5)
+    ));
+  }
+
+  /**
+   * Animator to loop the Flare animation. Use the LOOP behavior so that it's always shrinking bullseye rings.
+   */
+  private final Animator animator = new Animator(0, 1, Animator.RepeatBehavior.LOOP, new TimingTargetAdapter() {
+    @Override
+    public void begin() {
+      active = true;
+      animfrac = 0.0f;
+      repaintArea();
+    }
+
+    /**
+     * Animator tells us when to update the image.
+     * @param fraction Animator lets us know how far we are through our cycle
+     */
+    @Override
+    public void timingEvent(float fraction) {
+      animfrac = fraction;
+      repaintArea();
+    }
+
+    /**
+     * Animator tells us we're done.
+     */
+    @Override
+    public void end() {
+      active = false;
+      repaintArea();
+    }
+  });
+
+  /**
+   * Start the Flare animation.
+   * @param isLocal - true if this Flare was launched on this client (i.e. by this player). Otherwise, center on
+   *                  the ping location if user preferences are set for centering on opponent moves.
+   */
+  public void startAnimation(final boolean isLocal) {
+    if (map == null) {
+      return; // Means we already mentioned BadModuleData earlier. Now we simply avoid crashing.
+    }
+
+    if (!isLocal) {
+      if (GlobalOptions.getInstance().centerOnOpponentsMove()) {
+        map.centerAt(clickPoint);
+      }
+    }
+
+    animator.stop();
+    animate = pulses > 0 && pulsesPerSec > 0;
+    animator.setRepeatCount(Math.max(pulses, 1));
+    animator.setDuration(1000 / Math.max(pulsesPerSec, 1));
+    animator.start();
+  }
+
+  /**
+   * Draw the Flare at current animation frame
+   * @param g - Graphics
+   * @param map - Map component
+   */
+  public void draw(final Graphics g, final Map map) {
+    if (!active || (clickPoint == null)) {
+      return;
+    }
+    final Graphics2D g2d = (Graphics2D) g;
+    os_scale = g2d.getDeviceConfiguration().getDefaultTransform().getScaleX();
+
+    double diameter = (circleScale ? map.getZoom() : 1.0) * os_scale * circleSize;
+    if (animate) {
+      diameter *= (1.0 - animfrac);
+    }
+
+    if (diameter <= 0.0) {
+      return;
+    }
+
+    // translate the click location for current zoom
+    final Point p = map.mapToDrawing(clickPoint, os_scale);
+    g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+            RenderingHints.VALUE_ANTIALIAS_ON);
+
+    // draw a circle around the selected point
+    g2d.setColor(color);
+    g2d.setStroke(new BasicStroke((float)(STROKE * os_scale)));
+    g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+            RenderingHints.VALUE_ANTIALIAS_ON);
```

Comment:
No need to set the hints twice. Kill the earlier one?

Reply:
oh that's funny